### DISCUSSION TITLE
`prisma migrate dev` prompts me to reset my database, but I just baselined.

### DISCUSSION THREAD

**Discussion by emilbonnek** [prisma/prisma#19185] [Q&A]
**What did I do?**
I have been following this guide https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project/relational-databases-typescript-postgres (but using mysql instead of postgresql).

1. I entered my remote development database credentials in environment variables and ran `npx prisma db pull`. I confirmed that my schema prisma file is updated with the schema from my db.
2. I ran `npx prisma migrate diff --from-empty --to-schema-datamodel prisma/schema.prisma --script > prisma/migrations/0_init/migration.sql` to create an initial migration. I confirmed that the file exists in the `prisma/migrations/0_init` folder and that the sql looks right.
3. I ran `npx prisma migrate resolve --applied 0_init` and confirmed that the `_prisma_migrations` table is created in the database and that it has one record.
4. I made a minor change to my schema prisma file (made two fields non-nullable by removing the `?`'s at the end of the names).
5. I ran `npx prisma migrate dev --preview-feature --name make-values-non-nullable`

I am then met with this:
```
Drift detected: Your database schema is not in sync with your migration history.

The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.

It should be understood as the set of changes to get from the expected schema to the actual schema.

[+] Added tables
  - table_1
  - table_2
  - table_3
(All my tables are listed here)

? We need to reset the MySQL database "db_name" at "db_host:3306"
Do you want to continue? All data will be lost. » (y/N)
```

**What have I done?**
- Confirmed the user has the correct permissions.
- Setting up an explicit shadow db connection url (didn't seem to make any difference).
- Change line endings on the init migration file before running `npx prisma migrate resolve --applied 0_init` (I am on a Windows and I thought maybe line endings was messing up the hash when appliying the migration or something)
- Running `npx prisma migrate dev ...` before having made any changes. It still says it needs to reset my database, which makes me think the migration hasn't been applied?
- Searched discussions and asked chatGPT.

Any help greatly appreciated!

**Comment by ludralph** [comment]
Hi @emilbonnek 

Can you please look at this [guide](https://www.prisma.io/docs/guides/migrate/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute) for fixing failed migrations using `prisma migrate diff` command?

**Reply by nurul3101** [reply]
@emilbonnek Did you had a chance to look at Raphael's response? Please let us know in case you still need any help. If not, please consider marking the discussion as answered/closed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by mrigangka2003** [comment]
It happens because there might be some data in your database already but not present in local database and hence its having conflict to migrate it ! 

you may try npx prisma migrate reset to delete all the data and then try to 
run the command
`npx prisma migrate dev --name  nameof`
